File: greedy/lemonade-change.js
/**
 * @param {number[]} bills
 * @return {boolean}
 */
var lemonadeChange = function(bills) {
    let n = bills.length;
    let count5 = 0;
    let count10 = 0;
    let count20 = 0;
    for(let i =0;i<n;i++){
        if(bills[i] === 5) count5 = count5 + 1;
        else if(bills[i] === 10){
           if(count5 >= 1) {
            count10 = count10 + 1;
            count5 = count5-1;
           }
           else return false;
        }
        else{
            if(count5>=1 && count10>=1){
                count20 = count20+1;
                count10 = count10 -1;
                count5 = count5-1;
            }
            else{
                return false;
            }
        }
    }
    return true;
};

File: greedy/lemonade-change.cpp
class Solution {
public:
    bool lemonadeChange(vector<int>& bills) {
      int n = bills.size();

      int c5 = 0;
      int c10 = 0;

      for(int i =0;i<n;i++){
        if(bills[i] == 20){
            if(c5 >=1 && c10>=1 ){
                c5--;
                c10--;
            }
            else if(c5>=3) c5-=3;
            else return false;
        }
        else if(bills[i] == 10){
            if(c5 >= 1){
                c5--;
                c10++;
            }
            else return false;
        }
        else{
            c5++;
        }
      }  
      return true;
    }
};